From e1dffe0228892d631a70c38a83c6007c135b4f68 Mon Sep 17 00:00:00 2001
From: akodanka <anoob.anto.kodankandath@intel.com>
Date: Thu, 22 Oct 2020 13:03:32 +0530
Subject: [PATCH 2/2] Changes to fix compatibility issues with ngraph ops

Fix 1 : The eltwise add op for bias causes crash due to parallel
APIs called in mkldnn_eltwise_node.cpp.
Hence, enabled legacy _WIN32 for this file to prevent it.

Fix 2 : The transpose op causes crash in omp_get_max_threads.
Hence, changed THR from OMP to SEQ.

Tracked-On: OAM-92051
Signed-off-by: akodanka <anoob.anto.kodankandath@intel.com>
---
 Android.bp                                                    | 4 ++--
 .../src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp           | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Android.bp b/Android.bp
index aca95a75..a4f06b93 100644
--- a/Android.bp
+++ b/Android.bp
@@ -852,8 +852,8 @@ cc_library_shared {
         "-D__ANDROID__",
         "-DENABLE_MKL_DNN=1",
         "-DIE_BUILD_POSTFIX=\"\"",
-        "-DIE_THREAD=IE_THREAD_OMP",
-        "-DMKLDNN_THR=MKLDNN_THR_OMP",
+        "-DIE_THREAD=IE_THREAD_SEQ",
+        "-DMKLDNN_THR=MKLDNN_THR_SEQ",
         "-DNGRAPH_VERSION=\"0.0.0+a9c6e72\"",
         "-DIMPLEMENT_INFERENCE_ENGINE_PLUGIN",
         "-DHAVE_SSE42=1",
diff --git a/inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp b/inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp
index 50ec7d6c..5f4eac06 100644
--- a/inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp
+++ b/inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp
@@ -18,6 +18,7 @@
 #include "jit_uni_eltwise.hpp"
 #include "jit_uni_quantization.hpp"
 
+#define _WIN32 1
 using namespace mkldnn;
 using namespace MKLDNNPlugin;
 using namespace InferenceEngine;
-- 
2.17.1

